ACE/ace/CDR_Base.inl

```diff
@@ -145,7 +145,11 @@ ACE_CDR::swap_8 (const char* orig, char* target)
     && !defined(ACE_LACKS_INLINE_ASSEMBLY)
   ACE_REGISTER unsigned long x =
     * reinterpret_cast<const unsigned long*> (orig);
+#if defined (__MINGW64__)
```

Comment:
From what I can tell on MinGW-W64 targeting x86_64 (no matter the host), sizeof(long) == 4.  An 8-byte swap is not going to work on a long pointer unless it's explicitly qualified as "bswapq" instead of "bswap" at the assembly level.